docs/root/configuration/http/http_filters/rate_limit_quota_filter.rst

```diff
@@ -40,12 +40,25 @@ has expired, the filter can be configured to continue using the last quota assig
 The rate limit quota filter reports the request load for each bucket to the RLQS with the configured ``reporting_interval``. The RLQS may rebalance quota assignments based on the request
 load that each Envoy receives and push new quota assignments to Envoy instances.
 
-When the connection to RLQS server fails, the filter will fall back to either the
+Failure modes
+^^^^^^^^^^^^^
+
+In case the connection to RLQS server fails, the filter will fall back to either the
 :ref:`no assignment behavior <envoy_v3_api_field_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaBucketSettings.no_assignment_behavior>`
 if it has not yet received a rate limit quota or to the
 :ref:`expired assignment behavior <envoy_v3_api_field_extensions.filters.http.rate_limit_quota.v3.RateLimitQuotaBucketSettings.expired_assignment_behavior>` if
 connection could not be re-established by the time the existing quota expired.
 
+In case RLQS client, for any reason (including RLQS server connection failures), doesn't receive
+the initial bucket assignment within predetermined [2]_ time such bucket eventually will be purged
+from the memory. Any enqueued [1]_ requests will be rejected (only applies when "no assignment behavior"
+is to enqueue). Subsequent requests matched into the bucket will re-initialize the bucket in the "no assignment"
+state, restarting the reports. This is explained in more details at
+:ref:`Rate Limit Quota Service (RLQS) <envoy_v3_api_file_envoy/service/rate_limit_quota/v3/rlqs.proto>`.
+
+.. [1] Enqueuing (blocking) requests while waiting for the initial assigned is not currently supported.
```

Comment:
Why isn't this supported?  It seems like we need to do this before we remove the "work_in_progress" annotation, because otherwise control planes won't know whether they can set this for clients.

Reply:
This was originally a part of the design, but removed because there's no plans to implement it in the first iterations. The discussion is buried in the original PR, here's the screenshot

![image](https://github.com/envoyproxy/envoy/assets/672669/9ad56166-8acb-42c4-8446-7c6f7c18170b)

The change itself: https://github.com/envoyproxy/envoy/commit/f7665e145b75fb886c124ed6beaa1df35cd9fd8a

We probably should just remove all references to queuing from the rst.

Reply:
Removed in https://github.com/envoyproxy/envoy/pull/30581/commits/51b9abc042f14d7ed027b6eea89de7e34d1fda5d.